src/agents/policy_retriever_agent.py

```diff
@@ -0,0 +1,135 @@
+
+class PolicyRetrieverAgent:
+
+    def __init__(self, melding):
+        self.melding = melding
+
+    def retrieve_policy(self):
+        summarized_melding = self.summarize_melding()
+        context_text, sources = self.search_db(summarized_melding)
+        prompt = self.create_prompt(context_text)
+        response_text = self.invoke_model(prompt)
+        formatted_response = self.format_response(response_text, sources)
+        return formatted_response
+
+    def summarize_melding(self):
+        summarize_prompt = ChatPromptTemplate.from_template(cfg.SUMMARIZE_MELDING_TEMPLATE)
+        prompt = summarize_prompt.format(
+            melding=self.melding,
+        )
+        summarized_query = self.invoke_model(prompt, summarize=True)
+        
+        return summarized_query
+
+    def search_db(self, summarized_melding):
+        """
+        Search the Chroma database using the summarized melding,
+        combining and deduplicating results.
+
+        Args:
+            summarized_melding (str): The summarized melding text.
+
+        Returns:
+            tuple: A tuple containing:
+                - context_text (str): Combined context text from the search results.
+                - sources (list): List of sources used in the search.
+        """
+        db = Chroma(persist_directory=cfg.CHROMA_PATH, embedding_function=OpenAIEmbeddingFunction())
+
+        summarized_results = (
+            db.similarity_search_with_score(summarized_melding, k=5) if summarized_melding else []
+        )
+
+        combined_results = {
+            doc.metadata.get("source"): doc for doc, _ in summarized_results
+        }
+        sources = combined_results.keys()
+        sources = [source.split('/')[-1] for source in sources]
+        context_text = "\n\n---\n\n".join([open(os.path.join(cfg.DOCUMENTS_PATH, source), "r").read() for source in sources])
+        
+        return context_text, sources
+
+    def create_prompt(self, context_text):
+        prompt_template = ChatPromptTemplate.from_template(cfg.POLICY_MELDING_TEMPLATE)
+        prompt = prompt_template.format(
+            context=context_text,
+            melding=self.melding,
+        )
+        return prompt
+
+
+    def invoke_model(self, prompt, summarize=False):
+        """
+        Invoke the language model to generate a response based on the provided prompt.
+
+        Args:
+            prompt (str): The formatted prompt to be sent to the language model.
+            summarize (bool, optional): If True, use a summarizing system message. Defaults to False.
+
+        Returns:
+            str: The generated response from the language model.
+        """
+        system_content = "Je bent een behulpzame assistent" if summarize else "Je bent een behulpzame ambtenaar."
+
+        if cfg.ENDPOINT == 'local':
+            client = OpenAI(api_key=cfg.API_KEYS["openai"])
+        elif cfg.ENDPOINT == 'azure':
+            client = AzureOpenAI(
```

Comment:
This is the clients we were discussing yesterday in the meeting. 
I think all these clients should be taken out - you set up a client once at starting up the whole app and then just pass the client around for inference. In this way:
- you avoid these hardcoded model stuff all over the code base (instead, you have a single piece that handles all LLM logic and you can just swap the underlying model on one place)
- you avoid multiple initializations of the client (which probably also does authentication requests and things like that every time..)

Reply:
I'll work on centralizing all client initialization in a single file. However, similar to my earlier comment about `chromadb`, I believe it's preferable, at least for now, to initialize the client both during the frontend launch and within the CentralAgent itself. This approach allows us to test the CentralAgent independently of the frontend, enhancing flexibility during development.

Reply:
Ah sorry, miscommunication (my bad). 
I was just trying to show which clients we meant in the discussion yesterday, where we also said I can maybe fix that within the scope of the open source ticket.
So actually no worries, no need to change it, more like an FYI that I will probably get it out of there and some explanaton about why I will do so :)